packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/description/ImageDescription.java

```diff
@@ -0,0 +1,157 @@
+        return new Builder(id);
+    }
+
+    @Override
+    public String toString() {
+        String pattern = "{0} '{'id: {1}'}'"; //$NON-NLS-1$
+        return MessageFormat.format(pattern, this.getClass().getSimpleName(), this.getId());
+    }
+
+    /**
+     * Builder used to create the Image description.
+     *
+     * @author pcdavid
+     */
+    @SuppressWarnings("checkstyle:HiddenField")
+    public static final class Builder {
+        private final String id;
+
+        private Function<VariableManager, String> idProvider;
+
+        private Function<VariableManager, String> labelProvider;
+
+        private Function<VariableManager, String> iconURLProvider = variableManager -> null;
+
+        private Function<VariableManager, String> urlProvider;
+
+        private Function<VariableManager, String> maxWidthProvider = variableManager -> null;
+
+        private Function<VariableManager, List<?>> diagnosticsProvider;
+
+        private Function<Object, String> kindProvider;
+
+        private Function<Object, String> messageProvider;
+
+        private Builder(String id) {
+            this.id = Objects.requireNonNull(id);
+        }
+
+        public Builder idProvider(Function<VariableManager, String> idProvider) {
+            this.idProvider = Objects.requireNonNull(idProvider);
+            return this;
+        }
+
+        public Builder labelProvider(Function<VariableManager, String> labelProvider) {
+            this.labelProvider = Objects.requireNonNull(labelProvider);
+            return this;
+        }
+
+        public Builder iconURLProvider(Function<VariableManager, String> iconURLProvider) {
+            this.iconURLProvider = Objects.requireNonNull(iconURLProvider);
+            return this;
+        }
+
+        public Builder urlProvider(Function<VariableManager, String> urlProvider) {
+            this.urlProvider = Objects.requireNonNull(urlProvider);
+            return this;
+        }
+
+        public Builder maxWidthProvider(Function<VariableManager, String> maxWidthProvider) {
+            this.maxWidthProvider = Objects.requireNonNull(maxWidthProvider);
+            return this;
+        }
+
+        public Builder diagnosticsProvider(Function<VariableManager, List<?>> diagnosticsProvider) {
+            this.diagnosticsProvider = Objects.requireNonNull(diagnosticsProvider);
+            return this;
+        }
+
+        public Builder kindProvider(Function<Object, String> kindProvider) {
+            this.kindProvider = Objects.requireNonNull(kindProvider);
+            return this;
+        }
+
+        public Builder messageProvider(Function<Object, String> messageProvider) {
+            this.messageProvider = Objects.requireNonNull(messageProvider);
+            return this;
+        }
+
+        public ImageDescription build() {
+            ImageDescription imageDescription = new ImageDescription();
+            imageDescription.id = this.id;
```

Comment:
Please add Objects.requireNonNull(this.id)

Reply:
done